src/main/java/ch/njol/skript/effects/EffDetonate.java

```diff
@@ -34,47 +34,47 @@ public class EffDetonate extends Effect {
 		Skript.registerEffect(EffDetonate.class, "detonate %entities/blocks%");
 	}
 
-	private Expression<?> objects;
+	private Expression<?> thingsToDetonate;
 
 
 	@Override
 	@SuppressWarnings("unchecked")
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
-		this.objects = (Expression<?>) exprs[0];
+		this.thingsToDetonate = (Expression<?>) exprs[0];
  		return true;
 	}
 
 	@Override
 	protected void execute(Event event) {
-		for (Object object : objects.getArray(event)) {
-			if (object instanceof Block && ((Block) object).getType() == Material.TNT) {
-				Block block = (Block) object;
+		for (Object detonateThings : thingsToDetonate.getArray(event)) {
```

Comment:
`detonateThings` doesnt make sense because this is just a single object

Reply:
oops you're right fixing now